src/js/popper.js

```diff
@@ -0,0 +1,1210 @@
+    return e && e.referenceNode ? e.referenceNode : e;
+  }
+  function r(e) {
+    return 11 === e ? re : 10 === e ? pe : re || pe;
+  }
+  function p(e) {
+    if (!e) return document.documentElement;
+    for (
+      var o = r(10) ? document.body : null, n = e.offsetParent || null;
+      n === o && e.nextElementSibling;
+
+    )
+      n = (e = e.nextElementSibling).offsetParent;
+    var i = n && n.nodeName;
+    return i && "BODY" !== i && "HTML" !== i
+      ? -1 !== ["TH", "TD", "TABLE"].indexOf(n.nodeName) &&
+        "static" === t(n, "position")
+        ? p(n)
+        : n
+      : e
+      ? e.ownerDocument.documentElement
+      : document.documentElement;
+  }
+  function s(e) {
+    var t = e.nodeName;
+    return "BODY" !== t && ("HTML" === t || p(e.firstElementChild) === e);
+  }
+  function d(e) {
+    return null === e.parentNode ? e : d(e.parentNode);
+  }
+  function a(e, t) {
+    if (!e || !e.nodeType || !t || !t.nodeType) return document.documentElement;
+    var o = e.compareDocumentPosition(t) & Node.DOCUMENT_POSITION_FOLLOWING,
+      n = o ? e : t,
+      i = o ? t : e,
+      r = document.createRange();
+    r.setStart(n, 0), r.setEnd(i, 0);
+    var l = r.commonAncestorContainer;
+    if ((e !== l && t !== l) || n.contains(i)) return s(l) ? l : p(l);
+    var f = d(e);
+    return f.host ? a(f.host, t) : a(e, d(t).host);
+  }
+  function l(e) {
+    var t =
+        1 < arguments.length && void 0 !== arguments[1] ? arguments[1] : "top",
+      o = "top" === t ? "scrollTop" : "scrollLeft",
+      n = e.nodeName;
+    if ("BODY" === n || "HTML" === n) {
+      var i = e.ownerDocument.documentElement,
+        r = e.ownerDocument.scrollingElement || i;
+      return r[o];
+    }
+    return e[o];
+  }
+  function f(e, t) {
+    var o = 2 < arguments.length && void 0 !== arguments[2] && arguments[2],
+      n = l(t, "top"),
+      i = l(t, "left"),
+      r = o ? -1 : 1;
+    return (
+      (e.top += n * r),
+      (e.bottom += n * r),
+      (e.left += i * r),
+      (e.right += i * r),
+      e
+    );
+  }
+  function m(e, t) {
+    var o = "x" === t ? "Left" : "Top",
+      n = "Left" == o ? "Right" : "Bottom";
+    return (
+      parseFloat(e["border" + o + "Width"], 10) +
+      parseFloat(e["border" + n + "Width"], 10)
+    );
+  }
+  function h(e, t, o, n) {
+    return ee(
+      t["offset" + e],
+      t["scroll" + e],
+      o["client" + e],
+      o["offset" + e],
+      o["scroll" + e],
+      r(10)
+        ? parseInt(o["offset" + e]) +
+            parseInt(n["margin" + ("Height" === e ? "Top" : "Left")]) +
+            parseInt(n["margin" + ("Height" === e ? "Bottom" : "Right")])
+        : 0
+    );
+  }
+  function c(e) {
+    var t = e.body,
+      o = e.documentElement,
+      n = r(10) && getComputedStyle(o);
+    return { height: h("Height", t, o, n), width: h("Width", t, o, n) };
+  }
+  function g(e) {
+    return le({}, e, { right: e.left + e.width, bottom: e.top + e.height });
```

Comment:
Codacy found an issue: ['le' was used before it was defined.](https://app.codacy.com/manual/hello_100/jca-ui-kit/pullRequest?prid=5731813)